l10n_es_aeat_mod349/models/mod349.py

```diff
@@ -27,157 +28,176 @@ class Mod349(models.Model):
         return True
 
-    def _get_domain(self):
-        self.ensure_one()
-        domain = [('state', 'in', ['open', 'paid']),
-                  ('period_id', 'in', self.periods.ids),
-                  ('operation_key', '!=', False)]
-
-        if self.calculate_date:
-            year = fields.Date.from_string(self.fiscalyear_id.date_start)\
-                .year
-            if self.period_type == '0A':
-                date_start = "%s-01-01" % (year)
-                date_end = "%s-12-31" % (year)
-            elif self.period_type in ('1T', '2T', '3T', '4T'):
-                start_month = (int(self.period_type[:1]) - 1) * 3 + 1
-                date_start = "%s-%s-01" % (year, start_month)
-                date_end = "%s-%s-%s" % (year, start_month+2,
-                                         monthrange(year, start_month + 2)[1])
-            elif self.period_type in ('01', '02', '03', '04', '05', '06',
-                                      '07', '08', '09', '10', '11', '12'):
-                date_start = "%s-%s-01" % (year, self.period_type)
-                date_end = "%s-%s-%s" % (year, self.period_type,
-                                         monthrange(year,
-                                                    int(self.period_type))[1])
-            domain += [('date_invoice', '>=', date_start),
-                       ('date_invoice', '<=', date_end)]
-        return domain
-
     @api.multi
     def calculate(self):
         """Computes the records in report."""
+        partner_obj = self.env['res.partner']
         invoice_obj = self.env['account.invoice']
         for mod349 in self:
             # Remove previous partner records and partner refunds in report
             mod349.partner_record_ids.unlink()
             mod349.partner_refund_ids.unlink()
-            # Get corresponding invoices
-            domain = mod349._get_domain()
-            groups = invoice_obj.read_group(
-                domain, ['commercial_partner_id'], ['commercial_partner_id'])
-            for group in groups:
-                invoices_total = invoice_obj.search(group['__domain'])
-                invoices = invoices_total.filtered(
-                    lambda x: x.type in ('out_invoice', 'in_invoice'))
-                refunds = self.env['account.invoice']
-                # Filter refunds
-                for refund in invoices_total.filtered(
-                        lambda x: x.type in ('out_refund', 'in_refund')):
-                    if not refund.origin_invoices_ids:
-                        invoices += refund
-                        continue
-                    origin_inv = refund.origin_invoices_ids[0]
-                    if origin_inv.period_id not in mod349.periods:
-                        refunds += refund
-                    else:
-                        invoices += refund
+            # Returns all commercial partners
+            partners = partner_obj.with_context(active_test=False).search(
```

Comment:
This code is not really optimal, the loop will be executed on all partners which could cause a lack of perfomances on huge databases. Why did you replace the original read_group ?

Reply:
We are working on a new version for v9 to be forward-port it to v9. I'll tell when ready.